app/controllers/user_controller.rb

```diff
@@ -0,0 +1,7 @@
+class UserController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
I would suggest you put this in your application controller and any action that allows you not to be signed in do a skip before action. You are probably going to have way more actions that require you to be logged in than not logged in.